From: rubicon <rubicon@mailo.com>
Date: Thu, 02 Jun 2022 09:57:43 +0000
Subject: [PATCH] Fix compatibility of PL/Perl testsuite with Perl 5.36

Perl 5.36 moves the "Useless use of sort in scalar context" warning
from the old "void" category to a new "scalar" category. This causes
the PL/Perl testsuite to fail because the expected warning is not
produced when it sets warnings in the "void" category as FATAL.

This patch adds the "scalar" category to the list of FATAL warnings,
so the warning is produced as expected by the testsuite. The "void"
category needs to be kept for compatibility with older Perl versions.

Reference: https://perldoc.perl.org/perl5360delta#Changes-to-Existing-Diagnostics


--- a/src/pl/plperl/expected/plperl.out
+++ b/src/pl/plperl/expected/plperl.out
@@ -725,7 +725,7 @@
 CONTEXT:  PL/Perl anonymous code block
 -- check that we can "use warnings" (in this case to turn a warn into an error)
 -- yields "ERROR:  Useless use of sort in scalar context."
-DO $do$ use warnings FATAL => qw(void) ; my @y; my $x = sort @y; 1; $do$ LANGUAGE plperl;
+DO $do$ use warnings FATAL => qw(void scalar) ; my @y; my $x = sort @y; 1; $do$ LANGUAGE plperl;
 ERROR:  Useless use of sort in scalar context at line 1.
 CONTEXT:  PL/Perl anonymous code block
 -- make sure functions marked as VOID without an explicit return work
--- a/src/pl/plperl/sql/plperl.sql
+++ b/src/pl/plperl/sql/plperl.sql
@@ -470,7 +470,7 @@
 
 -- check that we can "use warnings" (in this case to turn a warn into an error)
 -- yields "ERROR:  Useless use of sort in scalar context."
-DO $do$ use warnings FATAL => qw(void) ; my @y; my $x = sort @y; 1; $do$ LANGUAGE plperl;
+DO $do$ use warnings FATAL => qw(void scalar) ; my @y; my $x = sort @y; 1; $do$ LANGUAGE plperl;
 
 -- make sure functions marked as VOID without an explicit return work
 CREATE OR REPLACE FUNCTION myfuncs() RETURNS void AS $$
